pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -110,6 +113,121 @@ type LbaasV2 struct {
 	LoadBalancer
 }
 
+// floatingSubnetSpec contains the specification of the public subnet to use for
+// a public network. If given it may either describe the subnet id or
+// a subnet name pattern for the subnet to use. If a pattern is given
+// the first subnet matching the name pattern with an allocatable floating ip
+// will be selected.
+type floatingSubnetSpec struct {
+	subnetID  string
+	subnet    string
+	subnetTag string
+}
+
+// matcher matches a subnet
+type matcher func(subnet *subnets.Subnet) bool
+
+// negate returns a negated matches for a given one
+func negate(f matcher) matcher { return func(s *subnets.Subnet) bool { return !f(s) } }
+
+func andMatcher(a, b matcher) matcher {
+	if a == nil {
+		return b
+	}
+	if b == nil {
+		return a
+	}
+	return func(s *subnets.Subnet) bool {
+		return a(s) && b(s)
+	}
+}
+
+// reexpNameMatcher creates a subnet matcher matching a subnet by name for a given regexp.
+func regexpNameMatcher(r *regexp.Regexp) matcher {
+	return func(s *subnets.Subnet) bool { return r.FindString(s.Name) == s.Name }
+}
+
+// subnetNameMatcher creates a subnet matcher matching a subnet by name for a given glob
+// or regexp
+func subnetNameMatcher(pat string) (matcher, error) {
+	// try to create floating IP in matching subnets
+	var match matcher
+	not := false
+	if strings.HasPrefix(pat, "!") {
+		not = true
+		pat = pat[1:]
+	}
+	if strings.HasPrefix(pat, "~") {
+		rexp, err := regexp.Compile(pat[1:])
+		if err != nil {
+			return nil, fmt.Errorf("invalid subnet regexp pattern %q: %s", pat[1:], err)
+		}
+		match = regexpNameMatcher(rexp)
+	} else {
+		match = regexpNameMatcher(glob.Globexp(pat))
+	}
+	if not {
+		match = negate(match)
+	}
+	return match, nil
+}
+
+// subnetTagMatcher matches a subnet by a given tag spec
+func subnetTagMatcher(tag string) matcher {
```

Comment:
gophercloud supports tags filtering using the neutron API: https://github.com/gophercloud/gophercloud/blob/bb95efe2e1e7225d263125171bed6f10770605ea/openstack/networking/v2/subnets/requests.go#L37-L40

Reply:
I added the tag filtering using the REST API